etcd-defrag/main.go

```diff
@@ -0,0 +1,105 @@
+package etcddefrag
+
+import (
+	"context"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/openshift/hypershift/pkg/version"
+	hyperapi "github.com/openshift/hypershift/support/api"
+	"github.com/openshift/hypershift/support/upsert"
+	"github.com/spf13/cobra"
+	"go.uber.org/zap/zapcore"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+)
+
+type Options struct {
+	Namespace     string
+	CertFile      string
+	KeyFile       string
+	TrustedCAFile string
+}
+
+func NewStartCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "etcd-defrag-controller",
+		Short: "Starts the etcd defrag controller",
+	}
+
+	opts := Options{
+		Namespace:     "",
+		CertFile:      "/var/run/secrets/etcd-client/etcd-client.crt",
+		KeyFile:       "/var/run/secrets/etcd-client/etcd-client.key",
+		TrustedCAFile: "/var/run/configmaps/etcd-ca/ca.crt",
```

Comment:
those options are ignored and not used by the controller. We should pass them to the etcdcli client instead of hardcoding them here
https://github.com/openshift/hypershift/pull/2869/files#diff-79a9561715e7c906edbb7c786fdbab2cc3f0e6cd83447a8a4aaa5c3c87b0aa2dR71-R75

Reply:
Oh ya, I forgot to plumb those in